ecosystem/lava-sdk/src/lavasession/consumerSessionManager.ts

```diff
@@ -786,61 +769,104 @@ export class ConsumerSessionManager {
+    epoch: number,
+    retry = 0
   ): Promise<any> {
+    if (retry != 0) {
+      await sleep(this.timeoutBetweenProbes());
+      if (this.currentEpoch != epoch) {
+        // incase epoch has passed we no longer need to probe old providers
+        Logger.info(
+          "during old probe providers epoch passed no need to query old providers."
+        );
+        return;
+      }
+    }
+
     Logger.debug(`providers probe initiated`);
-    let successfulProbeResponse: any = null;
+    const promiseProbeArray: Array<Promise<any>> = [];
+    const retryProbing: ConsumerSessionsWithProvider[] = [];
     for (const consumerSessionWithProvider of pairingList) {
       const startTime = performance.now();
-      try {
-        const probeResponse = await this.relayer.probeProvider(
-          consumerSessionWithProvider.endpoints[0].networkAddress,
-          this.getRpcEndpoint().apiInterface,
-          this.getRpcEndpoint().chainId
-        );
-        const endTime = performance.now();
-        const latency = endTime - startTime;
-        Logger.debug(
-          "Provider: " +
-            consumerSessionWithProvider.publicLavaAddress +
-            " chainID: " +
-            this.getRpcEndpoint().chainId +
-            " latency: ",
-          latency + " ms"
-        );
-
-        Logger.debug(
-          `providers probe done ${JSON.stringify({
-            endpoint: this.rpcEndpoint,
-          })}`
-        );
-
-        const lavaEpoch = probeResponse.getLavaEpoch();
-        Logger.debug(
-          `Lava Epoch for provider ${consumerSessionWithProvider.publicLavaAddress}: ${lavaEpoch}`
-        );
+      promiseProbeArray.push(
+        this.relayer
+          .probeProvider(
+            consumerSessionWithProvider.endpoints[0].networkAddress,
+            this.getRpcEndpoint().apiInterface,
+            this.getRpcEndpoint().chainId
+          )
+          .then((probeReply) => {
+            const endTime = performance.now();
+            const latency = endTime - startTime;
+            Logger.debug(
+              "Provider: " +
+                consumerSessionWithProvider.publicLavaAddress +
+                " chainID: " +
+                this.getRpcEndpoint().chainId +
+                " latency: ",
+              latency + " ms"
+            );
 
-        successfulProbeResponse = lavaEpoch;
-      } catch (err) {
-        console.log(
-          "Error while probing provider:",
-          consumerSessionWithProvider.publicLavaAddress
-        );
-        //Logger.error(err);
-      }
+            const lavaEpoch = probeReply.getLavaEpoch();
+            if (epoch == 0 && this.currentEpoch == 0) {
+              this.currentEpoch = lavaEpoch; // setting the epoch for initialization.
+            }
+            Logger.debug(
+              `Lava Epoch for provider ${consumerSessionWithProvider.publicLavaAddress}: ${lavaEpoch}`
+            );
+            this.epochTracker.setLatestBlockFor(
```

Comment:
Here we are missing:
```
consumerSessionWithProvider.setPairingEpoch(lavaEpoch);
```
If we do not setup pairingEpoch it will always be 0, so we need to to update it

Reply:
Which makes sense, when we probe the provider we have a lava epoch block, which should be used as the pairing epoch for this consumerSessionWithProvider. And not the one we set on init